Mathlib/Data/Matroid/IndepAxioms.lean

```diff
@@ -0,0 +1,560 @@
+      obtain ⟨J, ⟨hJ, hIJ, hJX⟩, hJmax⟩ := h_maximal X hXE I (h_subset hB.1 hIB) hIX
+      obtain ⟨BJ, hBJ⟩ := h_maximal E rfl.subset J hJ (h_support J hJ)
+      refine' ⟨J, ⟨⟨BJ,_, hBJ.1.2.1⟩ ,hIJ,hJX⟩, _⟩
+      · exact ⟨hBJ.1.1, fun B' hB' hBJB' ↦ hBJ.2 ⟨hB',hBJ.1.2.1.trans hBJB', h_support _ hB'⟩ hBJB'⟩
+      simp only [maximals, mem_setOf_eq, and_imp, forall_exists_index]
+      rintro A B' (hBi' : Indep _) - hB'' hIA hAX hJA
+      simp only [mem_setOf_eq, and_imp] at hJmax
+      exact hJmax (h_subset hBi' hB'') hIA hAX hJA )
+  ( fun B hB ↦ h_support B hB.1 )
+
+@[simp] theorem ofIndep_apply (E : Set α) (Indep : Set α → Prop) (h_empty : Indep ∅)
+    (h_subset : ∀ ⦃I J⦄, Indep J → I ⊆ J → Indep I)
+    (h_aug : ∀⦃I B⦄, Indep I → I ∉ maximals (· ⊆ ·) (setOf Indep) →
+      B ∈ maximals (· ⊆ ·) (setOf Indep) → ∃ x ∈ B \ I, Indep (insert x I))
+    (h_maximal : ∀ X, X ⊆ E → ExistsMaximalSubsetProperty Indep X)
+    (h_support : ∀ I, Indep I → I ⊆ E) :
+    (Matroid.ofIndep E Indep h_empty h_subset h_aug h_maximal h_support).Indep = Indep := by
+  ext I
+  simp_rw [indep_iff_subset_base, Matroid.ofIndep, ofBase_apply, mem_maximals_setOf_iff]
+  refine' ⟨fun ⟨B, ⟨hBi, _⟩, hIB⟩ ↦ h_subset hBi hIB, fun h ↦ _⟩
+  obtain ⟨B, hB⟩ := h_maximal E rfl.subset I h (h_support I h)
+  simp_rw [mem_maximals_setOf_iff, and_imp] at hB
+  exact ⟨B, ⟨hB.1.1, fun J hJ hBJ ↦ hB.2 hJ (hB.1.2.1.trans hBJ) (h_support J hJ) hBJ⟩, hB.1.2.1⟩
+
+@[simp] theorem ofIndep_ground (E : Set α) (Indep : Set α → Prop) (h_empty : Indep ∅)
+    (h_subset : ∀ ⦃I J⦄, Indep J → I ⊆ J → Indep I)
+    (h_aug : ∀⦃I B⦄, Indep I → I ∉ maximals (· ⊆ ·) (setOf Indep) →
+      B ∈ maximals (· ⊆ ·) (setOf Indep) → ∃ x ∈ B \ I, Indep (insert x I))
+    (h_maximal : ∀ X, X ⊆ E → ExistsMaximalSubsetProperty Indep X)
+    (h_support : ∀ I, Indep I → I ⊆ E) :
+    (Matroid.ofIndep E Indep h_empty h_subset h_aug h_maximal h_support).E = E := rfl
+
+/-- An independence predicate satisfying the finite matroid axioms determines a matroid,
+  provided independence is determined by its behaviour on finite sets.
+  This fundamentally needs choice, since it can be used to prove that every vector space
+  has a basis. -/
+protected def ofIndepOfFinitary (E : Set α) (Indep : Set α → Prop)
+    (h_empty : Indep ∅)
+    (ind_mono : ∀ ⦃I J⦄, Indep J → I ⊆ J → Indep I)
+    (ind_aug : ∀ ⦃I J⦄, Indep I → I.Finite → Indep J → J.Finite → I.ncard < J.ncard →
+      ∃ e ∈ J, e ∉ I ∧ Indep (insert e I))
+    (h_compact : ∀ I, (∀ J, J ⊆ I → J.Finite → Indep J) → Indep I)
+    (h_support : ∀ ⦃I⦄, Indep I → I ⊆ E) : Matroid α :=
+
+  have htofin : ∀ I e, Indep I → ¬ Indep (insert e I) →
+    ∃ I₀, I₀ ⊆ I ∧ I₀.Finite ∧ ¬ Indep (insert e I₀) := by
+      by_contra h; push_neg at h
+      obtain ⟨I, e, -, hIe, h⟩ := h
+      refine hIe <| h_compact _ fun J hJss hJfin ↦ ?_
+      exact ind_mono (h (J \ {e}) (by rwa [diff_subset_iff]) (hJfin.diff _)) (by simp)
+
+  Matroid.ofIndep E Indep h_empty ind_mono
+  ( by
```

Comment:
Please use keyword arguments here and elsewhere for `Matroid.ofIndep`
```suggestion
  (h_aug := by
```
Note that the `( by` on line 232 is too indented. (Also, just to mention it, the style would be `(by` instead of `( by`.)